kubernetes/kube-state-metrics-cluster-role.yaml

```diff
@@ -15,8 +18,11 @@ rules:
   verbs: ["list", "watch"]
 - apiGroups: ["extensions"]
   resources:
-  - daemonsets
   - deployments
+  verbs: ["list", "watch", "get", "update"]
```

Comment:
> In that case we need to separate the list/watch from the get and update as list/watch needs to be on all namespaces for the purpose of collecting the objects to produce the metrics of those. Thus get and update only need to be on that specific namespace.

exactly, in a Role in the `kube-system` namespace, bound with a `RoleBinding`. the rule should also specify `resourceNames:["kube-state-metrics"]` to only allow modifying the relevant deployment.